### PR TITLE
rgw:user email can modify to empty

### PR DESCRIPTION
**Pull request by dwj192**
user email can be empty in default
but can't  modify to empty when it has value

now it can modify to empty when need

Signed-off-by: Weijun Duan <duanweijun@h3c.com>

### CODE REVIEW

**Comment by dachary** [comment]
@dwj192 please ignore the bot failure, it is a false negative due to http://tracker.ceph.com/issues/13176

**Comment by yehudasa** [comment]
@dwj192 can you open a ceph tracker issue for that and note it in the commit message? Thanks

**Comment by dwj192** [comment]
@yehudasa  done! Thanks

**Comment by dwj192** [comment]
@yehudasa 
is it right now? thanks!

**Comment by dwj192** [comment]
@yehudasa   had delete email info in pool .users.email after set email to ""

**Comment by dwj192** [comment]
@yehudasa  my test
// create user
root@compiler:/home/dwj/git/email1/src#./radosgw-admin user create --uid=keyuser --display-name=keysuer--access-key=keyuser --secret=keyuser
{
   "user_id": "keyuser",
   "display_name": "keysuer",
   "email": "",
   ...
}

// add email for user
root@compiler:/home/dwj/git/email1/src#./radosgw-admin user modify --uid=keyuser --email=dwj@h3c.com
{
   "user_id": "keyuser",
   "display_name": "keysuer",
   "email": "dwj@h3c.com",
   ...
}

// look .users.email pool exist email info
root@compiler:/home/dwj/git/email1/src#./rados ls -p .users.email
dwj@h3c.com
s3@example.com
john.doe@example.com
tester@ceph.com

// modify user other info, user stile，email info still exist
root@compiler:/home/dwj/git/email1/src#./radosgw-admin user modify --uid=keyuser --display-name=keyuser1
{
   "user_id": "keyuser",
   "display_name": "keyuser1",
   "email": "dwj@h3c.com",
   ...
}

//  rm email info, email is empty
root@compiler:/home/dwj/git/email1/src#./radosgw-admin user modify --uid=keyuser --email=""
{
   "user_id": "keyuser",
   "display_name": "keyuser1",
   "email": "",
    ...
}

// look .users.email  pool  email is not exist
root@compiler:/home/dwj/git/email1/src#./rados ls -p .users.email
s3@example.com
john.doe@example.com
tester@ceph.com

**Comment by liewegas** [comment]
rebase please

**Comment by oritwas** [comment]
@dwj192 , your patch doesn't compile on master, can you update it?

File: src/rgw/rgw_user.cc [hunk 1]
```diff
@@ -2048,8 +2048,17 @@ int RGWUser::execute_modify(RGWUserAdminOpState& op_state, std::string *err_msg)
       }
     }
     user_info.user_email = op_email;
-  }
+  } else if (op_email.empty() && op_state.user_email_specified) {
 
+    rgw_obj email_obj(store->zone.user_email_pool, user_info.user_email);
+    ldout(store->ctx(), 10) << "removing email index: " << user_info.user_email << dendl;
+    ret = store->delete_system_obj(email_obj);
```
Comment (yehudasa) [inline src/rgw/rgw_user.cc]: Need to call `rgw_remove_email_index()` instead

**Comment by yehudasa** [comment]
@dwj192 closing this one in favor of this PR: https://github.com/ceph/ceph/pull/11340

**Pull request closed without merging**